src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -199,10 +202,18 @@ private CommandResult executeCommand(String commandText) throws CommandException
             }
 
             return commandResult;
-        } catch (CommandException | ParseException e) {
+        } catch (CommandException | ParseException | DataConversionException | IOException e) {
             logger.info("Invalid command: " + commandText);
             resultDisplay.setFeedbackToUser(e.getMessage());
             throw e;
         }
     }
+
+    /**
+     * Displays error message to user when there is invalid format in the teachersPet json file.
+     */
+    public void showInvalidFormatInJsonFile() {
+        resultDisplay.setFeedbackToUser("JSON file is invalid. File will not be saved until the error is corrected."
```

Comment:
I think it would be better to make it not a magical literal by declaring it as a private static variable.